lib/phase/functional.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const Joi = require('joi');
+const Hoek = require('hoek');
+const clone = require('clone');
+
+// Actual environment size is limited by space, not quantity.
+// We do this to force sd.yaml to be simpler.
+const MAX_ENVIRONMENT_VARS = 25;
```

Comment:
If there are 25 environment variables then for sure there will be more than 25 permutations. Either this number should be a lot lower, or MAX_PERMUTATION should be a lot higher. Or am I missing something?